File: programming_paradigms/object_oriented_programming/java_examples/ObjectOrientedAdvanced.java
package programming_paradigms.object_oriented_programming.java_examples;

import java.util.ArrayList;
import java.util.List;

class Library {
    private String name;
    private List<Book> books;

    public Library(String name) {
        this.name = name;
        this.books = new ArrayList<>();
    }

    public void addBook(Book book) {
        books.add(book);
    }

    public void displayBooks() {
        System.out.println("Books in " + name + " library:");
        for (Book book : books) {
            System.out.println(book.getTitle() + " by " + book.getAuthor());
        }
    }
}

class Book {
    private String title;
    private String author;

    public Book(String title, String author) {
        this.title = title;
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }
}

public class ObjectOrientedAdvanced {
    public static void main(String[] args) {
        Library library = new Library("City Library");

        Book book1 = new Book("The Great Gatsby", "F. Scott Fitzgerald");
        Book book2 = new Book("To Kill a Mockingbird", "Harper Lee");

        library.addBook(book1);
        library.addBook(book2);

        library.displayBooks();
    }
}

File: programming_paradigms/object_oriented_programming/golang_examples/ObjectOrientedAdvanced.go
package golangexamples

import "fmt"

type Book struct {
	Title  string
	Author string
}

type Library struct {
	Name  string
	Books []Book
}

func (l *Library) AddBook(book Book) {
	l.Books = append(l.Books, book)
}

func (l Library) DisplayBooks() {
	fmt.Printf("Books in %s library:\n", l.Name)
	for _, book := range l.Books {
		fmt.Printf("%s by %s\n", book.Title, book.Author)
	}
}

func main() {
	library := Library{Name: "City Library"}

	book1 := Book{Title: "The Great Gatsby", Author: "F. Scott Fitzgerald"}
	book2 := Book{Title: "To Kill a Mockingbird", Author: "Harper Lee"}

	library.AddBook(book1)
	library.AddBook(book2)

	library.DisplayBooks()
}
